presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java

```diff
@@ -522,7 +523,7 @@ private boolean isPushdownFilterSupported(ConnectorSession session, TableHandle
         boolean pushdownFilterEnabled = HiveSessionProperties.isPushdownFilterEnabled(session);
         if (pushdownFilterEnabled) {
             HiveStorageFormat hiveStorageFormat = getHiveStorageFormat(getMetadata(tableHandle).getTableMetadata(session, tableHandle.getConnectorHandle()).getProperties());
-            if (hiveStorageFormat == HiveStorageFormat.ORC || hiveStorageFormat == HiveStorageFormat.DWRF) {
+            if (hiveStorageFormat == HiveStorageFormat.ORC || hiveStorageFormat == HiveStorageFormat.DWRF || hiveStorageFormat == HiveStorageFormat.PARQUET && isParquetPushdownFilterEnabled(session)) {
```

Comment:
missing parentheses?

Reply:
THanks @mbasmanova for reviewing the PR.

> Is this the case? The planner would push down filter into table scan and remove the filter node above it. If the reader is not going to process the filter, the query results will be wrong. Shouldn't we add a check in the reader to fail fast if there is a pushed down filter that it cannot yet process?

My intention was that default value of `hive.parquet.pushdown_filter_enabled` shall remain false, but you were right, when it was turned on, we should fail the query early. I updated the PR and fail the query with NON_SUPPORTED exception in the newly introduced empty `ParquetSelectivePageSourceFactory`.

> It would be nice to add a test.

`testParquetSelectivePageSourceFails` was added to `TestHiveIntegrationSmokeTest.java`